### DISCUSSION TITLE
Getting HA pool/spa temp thermostats to operate in deg C

### DISCUSSION THREAD

**Discussion by romirez** [aqualinkd/AqualinkD#368] [General]
Hi, I'm trying to get aqualinkd to play nice with Home Assistant configured for celcius via mqtt, and having trouble getting the thermostat controls to cooperate. The sensors correctly display in C, but thermostat data is all in F (despite labels saying C)

<img width="805" alt="image" src="https://github.com/user-attachments/assets/258f3013-0a17-49c4-8061-c7389305797b">

I've tried every combination of F/C settings in Aqualink, and aqualinkd.conf, and when everything (including HA) is configured for F, everything works. When HA is in C and convert_mqtt_temp_to_c is set to "yes", I get the above behavior. Is there something I am missing?

**Comment by sfeakes** [comment]
That's something I had not tested.  Looks like a bug.  I've just fixed it, but it bought up other issues now since HA only seems to adjust the reporting units of measure when you do it this way and not all the other parts of the thermostat, like max min etc.  I'll post an update to 2.5.1(dev) later today hopefully.

**Comment by sfeakes** [comment]
I just posted an update to 2.5.1 (dev), if you re-get that dev version and try again.   I did notice that when you change HA from F to C you are best off restarting AqualinkD since that will force HA to accept a few changes.

**Comment by romirez** [comment]
Yup, looks like it's working now! Amazing - thank you for such a quick fix :)
